framework_tvb/tvb/interfaces/web/entities/context_simulator.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+#
+# TheVirtualBrain-Framework Package. This package holds all Data Management, and
+# Web-UI helpful to run brain-simulations. To use it, you also need do download
+# TheVirtualBrain-Scientific Package (for simulators). See content of the
+# documentation-folder for more details. See also http://www.thevirtualbrain.org
+#
+# (c) 2012-2020, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+
+"""
+.. moduleauthor:: Robert Vincze <robert.vincze@codemart.ro>
+"""
+
+from tvb.core.entities.file.simulator.view_model import SimulatorAdapterModel
+from tvb.core.entities.model.model_burst import BurstConfiguration
+from tvb.interfaces.web.controllers import common
+
+
+class SimulatorContext(object):
+    def __init__(self):
+        self.session_stored_simulator = None
```

Comment:
Why keep these attributes also on SimContext if we already keep them in session?

Reply:
I thought the point was to keep all session related parameters on the context to not have to get them from the session in each method where they  are used.  So I am a little confused now about how I should change the code.

Reply:
I don't think session access is a problem, you have to do it anyway, in each endpoint. 
Introducing this SimulatorContext would improve the code by handling all the necessary flags together, in one place, instead of having to handle each flag separately in each endpoint. So, instead of having multiple lines of code that get values from the session, you would have only one that calls a method on the SimulatorContext. For me, keeping these values both on the session and on the context seems a bit much since you always have to make sure they are synchronized.
What I would expect to have here are the keys from common.py and methods that add/get those to/from the session for different use-cases.